Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp

```diff
@@ -1617,6 +1637,16 @@ void NetworkProcessProxy::increaseQuota(PAL::SessionID sessionID, const WebCore:
     });
 }
 
+void NetworkProcessProxy::sendStorageAccessPromptQuirksData()
+{
+#if ENABLE(ADVANCED_PRIVACY_PROTECTIONS)
+    if (StorageAccessPromptQuirkController::shared().cachedQuirks().isEmpty())
```

Comment:
What if the quirks get updated so that we go from a non-empty list to an empty list? Isn't that a possibility (i.e. we no longer need the quirks). It seems we would fail to update the WebProcess's quirks in this case no?